packages/ui/src/primitives/tailwind/Select/index.tsx

```diff
@@ -23,181 +23,259 @@ All portions of the code written by the Infinite Reality Engine team are Copyrig
+import { CheckLg, ChevronDownSm, HelpIconSm } from '@ir-engine/ui/src/icons'
+import React, { useEffect, useLayoutEffect, useRef, useState } from 'react'
+import { twMerge } from 'tailwind-merge'
+import { InputProps, variantSizes } from '../Input'
+import Tooltip from '../Tooltip'
 
-import Input, { InputProps } from '../Input'
+export interface MenuItemProps {
+  children: React.ReactNode
+  value: string | number
+  disabled?: boolean
+  selected?: boolean
+  onClick?: () => void
+  showCheckmark?: boolean
+}
 
-export type OptionValueType = string | number
+export const MenuItem = ({ children, selected, disabled, onClick, showCheckmark }: MenuItemProps) => {
+  return (
+    <div
+      tabIndex={0}
+      className={twMerge(
+        `flex cursor-pointer items-center gap-x-2 bg-[#141619] px-4 py-2.5 text-xs text-[#9CA0AA] hover:bg-[#191B1F] hover:text-[#F5F5F5] ${
+          selected && 'text-[#375DAF]'
+        } ${disabled && 'text-[#42454D]'} w-full`
+      )}
+      onClick={onClick}
+    >
+      <div className="w-full">{children}</div>
 
-export type SelectOptionsType = { label: string; value: any; disabled?: boolean }
+      {selected && showCheckmark && <CheckLg color="#375DAF" />}
+    </div>
+  )
+}
 
-export interface SelectProps<T extends OptionValueType> {
-  label?: string
-  className?: string
-  error?: string
-  description?: string
-  options: SelectOptionsType[]
-  currentValue: T
+export interface SelectProps<T = string | number> {
+  children: React.ReactElement<typeof MenuItem> | React.ReactElement<typeof MenuItem>[]
+  width?: 'sm' | 'md' | 'lg' | 'full'
+  inputSizeVariant?: InputProps['variantSize']
   onChange: (value: T) => void
-  placeholder?: string
+  value: T
+  renderValue?: (value: T) => React.ReactNode
+  labelProps?: InputProps['labelProps']
+  state?: InputProps['state']
+  helperText?: InputProps['helperText']
+  required?: boolean
+  showCheckmark?: boolean
   disabled?: boolean
-  menuClassname?: string
-  menuItemClassName?: string
-  labelClassName?: string
-  inputVariant?: InputProps['variantSize']
-  inputClassName?: string
-  errorBorder?: boolean
-  searchDisabled?: boolean
-  inputContainerClassName?: string
-  endComponent?: JSX.Element
 }
 
-const Select = <T extends OptionValueType>({
-  className,
-  label,
-  error,
-  description,
-  options,
-  currentValue,
+const Select = ({
+  children,
+  width = 'md',
+  inputSizeVariant = 'l',
   onChange,
-  placeholder,
-  disabled,
-  menuClassname,
-  menuItemClassName,
-  labelClassName,
-  inputVariant,
-  inputClassName,
-  errorBorder,
-  searchDisabled,
-  inputContainerClassName,
-  endComponent
-}: SelectProps<T>) => {
+  value,
+  renderValue,
+  labelProps,
+  state,
+  helperText,
+  required,
+  showCheckmark,
+  disabled
+}: SelectProps) => {
+  const variantToWidth: Record<NonNullable<SelectProps['width']>, string> = {
+    sm: '240px',
+    md: '320px',
+    lg: '520px',
+    full: '100%'
+  }
```

Comment:
let's move this outside the select component but inside the same file

Reply:
Done